src/django_rich/test.py

```diff
@@ -0,0 +1,180 @@
+            self.console.print("expected failure", style=Style(color="yellow"))
+        elif self.dots:
+            self.console.print("x", style=Style(color="yellow"), end="")
+
+    @failfast
+    def addUnexpectedSuccess(self, test: TestCase) -> None:
+        self.unexpectedSuccesses.append(test)
+        if self.showAll:
+            self.console.print("unexpected success", style=Style(color="red"))
+        elif self.dots:
+            self.console.print("u", style=Style(color="red"), end="")
+
+    # Typeshed had wrong signature
+    # https://github.com/python/typeshed/pull/7341
+    def printErrorList(  # type: ignore [override]
+        self,
+        flavour: str,
+        errors: list[tuple[TestCase, str]],
+    ) -> None:
+        for test, err in errors:
+            rule = Rule(style=DJANGO_GREEN)
+            title = f"{flavour}: {self.getDescription(test)}"
+            self.console.print(rule, title, rule)
+            self.stream.write("%s\n" % err)
+
+    def _exc_info_to_string(self, err: _SysExcInfoType, test: TestCase) -> str:
+        """Converts a sys.exc_info()-style tuple of values into a string."""
+        exctype, value, tb = err
+        # Skip test runner traceback levels
+        while tb and self._is_relevant_tb_level(tb):  # type: ignore [attr-defined]
+            tb = tb.tb_next
+
+        msgLines = []
+        if exctype is not None:
+            # assert value is not None
+            # assert tb is not None
+            extract = Traceback.extract(exctype, value, tb, show_locals=True)
+            tb_e = Traceback(extract, suppress=[unittest, testcases])
+            with self.console.capture() as capture:
+                self.console.print(tb_e)
+            msgLines.append(capture.get())
+
+        if self.buffer:
+            assert isinstance(sys.stdout, io.StringIO)
+            output = sys.stdout.getvalue()
+            assert isinstance(sys.stderr, io.StringIO)
+            error = sys.stderr.getvalue()
+            if output:
+                if not output.endswith("\n"):
+                    output += "\n"
+                msgLines.append(STDOUT_LINE % output)
+            if error:
+                if not error.endswith("\n"):
+                    error += "\n"
+                msgLines.append(STDERR_LINE % error)
+
+        return "".join(msgLines)
+
+
+class RichDebugSQLTextTestResult(DebugSQLTextTestResult, RichTextTestResult):
+    # DebugSQLTextTestResult adds sql_debug to errors in type-incompatible way
+    def printErrorList(  # type: ignore [override]
+        self,
+        flavour: str,
+        errors: Iterable[tuple[TestCase, str, str]],
+    ) -> None:
+        for test, err, sql_debug in errors:
+            rule = Rule(style=DJANGO_GREEN)
+            title = f"{flavour}: {self.getDescription(test)}"
+            self.console.print(rule, title, rule)
+            self.console.print("%s\n" % err)
+            self.console.print(rule)
+            self.console.print(sql_debug)
```

Comment:
It feels like we should be assisting by letting Rich know we're are trying to output SQL. I tried earlier in the week but was struggling a little. Maybe it's because sql_debug isn't just SQL. 

https://rich.readthedocs.io/en/stable/syntax.html#syntax

Maybe it looks good enough with the current output. 🤔

Reply:
Yeah - I am not sure we can even parse out the SQL since it uses the logging framework and we may not be able to guarantee the logger format that would be used.

Currently it looks like:

<img width="761" alt="Screenshot 2022-02-24 at 18 58 22" src="https://user-images.githubusercontent.com/857609/155589612-b3e2a4ed-55ae-4244-9a63-e2719961d93e.png">

Rich's default highlighting does cover values.